xrpl/models/transactions/amm_bid.py

```diff
@@ -0,0 +1,138 @@
+"""Model for AMMBid transaction type."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type
+
+from typing_extensions import Final
+
+from xrpl.models.amounts import Amount
+from xrpl.models.base_model import BaseModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+_MAX_AUTH_ACCOUNTS: Final[int] = 4
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AuthAccount(BaseModel):
```

Comment:
Since this is also going to be exported, should it be in its own file in xrpl/models?

Reply:
Since we do this for all models, I'll make it its own file for consistency.

Reply:
Actually, `xrpl/models/transactions/signer_list_set.py` contains both BaseModel and Transaction in one file: `SignerEntry` and `SignerListSet`. So let's keep it in one file.

Reply:
Update: decided to put `AuthAccount` in its own file since this model is used for initializing `auth_accounts` in `AMMBid`